docs/source/user_guide/timeseries.ipynb

```diff
@@ -851,7 +851,128 @@
+    "\n",
+    "print(\"Dates needed:\")\n",
+    "print(f\"{beginning_date.strftime('%Y-%m-%d %X')} to {end_date.strftime('%Y-%m-%d %X')}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### We can see how the dates are valid by generating some data with the above date range."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
    "metadata": {},
+   "outputs": [],
+   "source": [
+    "import random\n",
+    "\n",
+    "dates = pd.date_range(\n",
+    "    beginning_date,\n",
+    "    end_date,\n",
+    "    freq=pl.frequency.split(\"-\")[0],\n",
+    ")\n",
+    "\n",
+    "X_train_forecast = pd.DataFrame(index=[i + 1 for i in range(len(dates))])\n",
+    "categorical_feature = pd.Series(\n",
+    "    [random.randint(0, 3) for i in range(len(dates))], index=X_train_forecast.index\n",
+    ")\n",
+    "numeric_feature = pd.Series(\n",
+    "    [i + 1 for i in range(len(dates))], index=X_train_forecast.index\n",
+    ")\n",
+    "\n",
+    "X_train_forecast[\"Date\"] = pd.Series(dates.values, index=X_train_forecast.index)\n",
+    "X_train_forecast[\"Categorical\"] = pd.Series(\n",
+    "    categorical_feature.values, index=X_train_forecast.index\n",
+    ")\n",
+    "X_train_forecast[\"Numeric\"] = pd.Series(\n",
+    "    numeric_feature.values, index=X_train_forecast.index\n",
+    ")\n",
+    "X_train_forecast.ww.init(\n",
+    "    logical_types={\"Categorical\": \"categorical\", \"Numeric\": \"integer\"}\n",
+    ")\n",
+    "\n",
+    "y_train_forecast = pd.Series(\n",
+    "    X_train_forecast[\"Numeric\"].values, index=X_train_forecast.index\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "X_test_forecast = pd.DataFrame(\n",
+    "    {\"Date\": [forecast_date], \"Categorical\": [3], \"Numeric\": [53862]}\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### ... and we succesfully have our prediction!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pl.predict(X_test_forecast, X_train=X_train_forecast, y_train=y_train_forecast)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "forecast_start = pd.Timestamp(\"1991-01-07\")\n",
+    "forecast_end = pd.Timestamp(\"1991-01-14\")\n",
+    "\n",
+    "dates = pl.dates_needed_for_prediction_range(forecast_start, forecast_end)\n",
+    "print(\"Dates needed:\")\n",
+    "print(f\"{dates[0].strftime('%Y-%m-%d %X')} to {dates[1].strftime('%Y-%m-%d %X')}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
```

Comment:
This cell feels out of place to me, I'm not quite sure why it's here. Is it supposed to be somewhere else?